test/unit/test_connection_factory.py

```diff
@@ -177,6 +188,181 @@ def test_get_saas_kwargs(
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_TRUSTED_CA,
+                env_value="my_ca_dir",
+                is_dir=True,
+                ssl_name="ca_cert_path",
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_TRUSTED_CA,
+                env_value="my_ca_file.cert",
+                is_file=True,
+                ssl_name="ca_certs",
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CLIENT_CERT,
+                env_value="my_cli_file.cert",
+                is_file=True,
+                ssl_name="certfile",
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CLIENT_CERT,
+                env_value="my_cli_file.cert",
+                is_file=True,
+                ssl_name="certfile",
+            ),
+            SslTestOpt(
+                env_name=ENV_SSL_PRIVATE_KEY,
+                env_value="my_private.key",
+                is_file=True,
+                ssl_name="keyfile",
+            ),
+        ],
+    ],
+    ids=[
+        "cert_reqs_required",
+        "cert_reqs_none",
+        "cert_reqs_unknown",
+        "ca_cert_path",
+        "ca_certs",
+        "certfile",
+        "keyfile",
+    ],
+)
+def test_get_ssl_options(test_opts, tmp_path) -> None:
+    env: dict[str, str] = {}
+    expected: dict[str, Any] = {}
+    for test_opt in test_opts:
+        env_value, ssl_value = test_opt.get_test_values(tmp_path)
+        if env_value is not None:
+            env[test_opt.env_name] = env_value
+        expected[test_opt.ssl_name] = ssl_value
+    ssl_opt = get_ssl_options(env)
+    for key, val in expected.items():
+        assert ssl_opt.get(key) == val
+
+
+@pytest.mark.parametrize(
+    "test_opts",
+    [
+        [SslTestOpt(env_name=ENV_SSL_TRUSTED_CA, env_value="my_ca_dir", is_dir=True)],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_TRUSTED_CA, env_value="my_ca_file.cert", is_file=True
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CLIENT_CERT, env_value="my_cli_file.cert", is_file=True
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CLIENT_CERT, env_value="my_cli_file.cert", is_file=True
+            ),
+            SslTestOpt(
+                env_name=ENV_SSL_PRIVATE_KEY, env_value="my_private.key", is_file=True
+            ),
+        ],
+    ],
+    ids=["ca_cert_path", "ca_certs", "certfile", "keyfile"],
+)
+def test_get_ssl_options_invalid(test_opts, tmp_path) -> None:
+    env: dict[str, str] = {}
+    last_env_name = ""
+    for test_opt in test_opts:
+        env_value, _ = test_opt.get_test_values(tmp_path)
+        last_env_name = test_opt.env_name
+        env[last_env_name] = env_value
+    env[last_env_name] = str(tmp_path / "non-existent")
```

Comment:
And here we actually make the env incorrect , get_test_values returns always a correct example. Hm, a bit confusing, the test should at least get a doc string